endpoints-control/src/main/java/com/google/api/control/model/ReportingRule.java

```diff
@@ -94,15 +91,16 @@ public static ReportingRule fromService(Service s) {
    */
   public static ReportingRule fromKnownInputs(@Nullable String[] logs,
       @Nullable Set<String> metricNames, @Nullable Set<String> labelNames) {
-    log.fine(String.format("creating rule from log names %s, metric names %s, labelNames %s",
-        Arrays.toString(logs), metricNames, labelNames));
+    log.atFine().log(
+        "creating rule from log names %s, metric names %s, labelNames %s",
+        Arrays.toString(logs), metricNames, labelNames);
```

Comment:
Can this `toString()` call be avoided here?

Reply:
Good call, I did use lazy here.